vendor/arbiter/arbiter.cpp

```diff
@@ -4735,8 +4725,20 @@ namespace ARBITER_CUSTOM_NAMESPACE
 
 namespace arbiter
 {
-namespace util
+
+namespace
+{
+    std::mutex randomMutex;
```

Comment:
Seems like all of this could be static if you're going to lock anyway, not that it matters much.